server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -388,7 +396,7 @@ public ClusterState execute(ClusterState currentState) {
                     shards,
                     userMeta,
                     version,
-                    featureStates
+                    new ArrayList<>(featureStates)
```

Comment:
Let's use List.copyOf() here as well, no need to mutable lists as ctor arguments here.